tests/integration/Mailer/WordPress/WordpressMailerTest.php

```diff
@@ -268,6 +275,131 @@ public function testItAddSubscriberMetaInfo() {
     $wpMailer->send();
   }
 
+  /*
+   * Test for issue https://mailpoet.atlassian.net/browse/MAILPOET-3707
+   */
+  public function testItPreservesWPReplyTo() {
+    list($mailer, $fallbackMailer) = $this->getMailerInstancesForReplyToTests();
+
+    $wpMailer = new WordPressMailer($mailer, $fallbackMailer, new MetaInfo, $this->subscribersRepository);
+    $wpMailer->addAddress('email@example.com', 'Full Name');
+    $wpMailer->addReplyTo('reply-to@example.com', 'Reply To');
+    $wpMailer->From = 'email-from@example.com';
+    $wpMailer->Body = 'Body';
+    $wpMailer->send();
+
+    $mailer = $this->getPrivateProperty($wpMailer, 'mailer');
+
+    $this->assertEquals('reply-to@example.com', $mailer->replyTo['reply_to_email']);
+    $this->assertEquals('Reply To', $mailer->replyTo['reply_to_name']);
+  }
+
+  /*
+   * Test for issue https://mailpoet.atlassian.net/browse/MAILPOET-3707
+   */
+  public function testItSetsSenderAsReplyToWhenReplyToIsNotDefined() {
+    list($mailer, $fallbackMailer) = $this->getMailerInstancesForReplyToTests();
+
+    $wpMailer = new WordPressMailer($mailer, $fallbackMailer, new MetaInfo, $this->subscribersRepository);
+    $wpMailer->addAddress('email@example.com', 'Full Name');
+    $wpMailer->From = 'email-from@example.com';
+    $wpMailer->Body = 'Body';
+    $wpMailer->send();
+
+    $mailer = $this->getPrivateProperty($wpMailer, 'mailer');
+
+    $this->assertEquals('staff@mailinator.com', $mailer->replyTo['reply_to_email']);
+    $this->assertEquals('Sender', $mailer->replyTo['reply_to_name']);
+  }
+
+  /*
+   * Test for issue https://mailpoet.atlassian.net/browse/MAILPOET-3707
+   */
+  public function testItChangesReplyToEmailOnDifferentCalls() {
+    list($mailer, $fallbackMailer) = $this->getMailerInstancesForReplyToTests();
+
+    $wpMailer = new WordPressMailer($mailer, $fallbackMailer, new MetaInfo, $this->subscribersRepository);
+    $wpMailer->addAddress('email@example.com', 'Full Name');
+    $wpMailer->addReplyTo('reply-to@example.com', 'Reply To');
+    $wpMailer->From = 'email-from@example.com';
+    $wpMailer->Body = 'Body';
+    $wpMailer->send();
+
+    $mailer = $this->getPrivateProperty($wpMailer, 'mailer');
+
+    $this->assertEquals('reply-to@example.com', $mailer->replyTo['reply_to_email']);
+    $this->assertEquals('Reply To', $mailer->replyTo['reply_to_name']);
+
+    $wpMailer = new WordPressMailer($mailer, $fallbackMailer, new MetaInfo, $this->subscribersRepository);
+    $wpMailer->addAddress('email@example.com', 'Full Name');
+    $wpMailer->From = 'email-from@example.com';
+    $wpMailer->Body = 'Body';
+    $wpMailer->send();
+
+    $mailer = $this->getPrivateProperty($wpMailer, 'mailer');
+
+    $this->assertEquals('staff@mailinator.com', $mailer->replyTo['reply_to_email']);
+    $this->assertEquals('Sender', $mailer->replyTo['reply_to_name']);
+  }
+
+  public function testWPEmailReplyToShouldntAffectOtherEmails() {
```

Comment:
This test demonstrates the current limitation of the approach proposed here, which I'm inclined to think is a dealbreaker. See the comments below for instructions on how to check the test results as it doesn't contain any assertion.